File: 11459/main.c
#include <stdio.h>
#include <stdlib.h>

int players[1000000];
int step[101];
int a, b, c, winner;

void init(){
    winner = 0;
    int i, x, p;
    for(i = 0; i < a; i++)
        players[i] = 1;

    for(i = 0; i < 101; i++)
        step[i] = i;


    for(i = 0; i < b; i++){
        scanf("%d %d", &x, &p);
        step[x] = p;
    }

}

void rolls(){
    int i, x, j;
    for(i = 0; i < c; i++)
    {
        scanf("%d", &x);
        if(!winner)
        {
            j = players[i%a] + x;
            if(step[j] < 100)
                players[i%a] = step[j];
            else
            {
                players[i%a] = 100;
                winner = 1;
            }
        }
    }
}

int main()
{
    int N, i;
    scanf("%d", &N);
    while(N--)
    {
        scanf("%d %d %d", &a, &b, &c);
        init();
        rolls();

        for(i = 0; i < a; i++)
            printf("Position of player %d is %d.\n", i + 1, players[i]);
    }
    return 0;
}

File: 11459/Main.java
import java.util.Scanner;

public class Main {
    static Scanner sc = new Scanner(System.in);
    static int[] players = new int[1000000];
    public static void main(String[] args) {
        int N, a, b, c, i, roll, current;
        boolean winner;
        N = Integer.parseInt(sc.nextLine());
        while(N > 0){
            winner = false;
            String[] data = sc.nextLine().split(" ");

            a = Integer.parseInt(data[0]);
            b = Integer.parseInt(data[1]);
            c = Integer.parseInt(data[2]);

            int[] step = new int[101];

            for(i = 0; i < a; i++)
                players[i] = 1;

            for(i = 0; i < 101; i++)
                step[i] = i;

            for(i = 0; i < b; i++){
                String[] snakeLadder = sc.nextLine().split(" ");
                step[Integer.parseInt(snakeLadder[0])] = Integer.parseInt(snakeLadder[1]);
            }

            for(i = 0; i < c; i++){
                roll = Integer.parseInt(sc.nextLine());
                if(!winner){
                    current = players[i % a] + roll;
                    if(step[current] < 100){
                        players[i % a] = step[current];
                    }
                    else{
                        players[i % a] = 100;
                        winner = true;
                    }
                }
            }

            for(i = 0; i < a; i++)
                System.out.println("Position of player " + (i + 1) + " is " + players[i] + ".");

            N--;
        }
    }
}
